### ISSUE TITLE
Mtproto and ad tag problem

### ISSUE THREAD

**Issue by AINNNNNNDEN** [alexbers/mtprotoproxy#304]
My mt proto proxy is working just fine but when I add a Ad Tag to it in config file it just stop working in official versian it keeps flickering connecting and connected
Im on Ubuntu 22
Using direct ip for proxy
Please advise

**Comment by alexbers** [comment]
then ad tag enabled, the proxy uses middle proxy. Please check the server has connectivity to servers specified here: https://core.telegram.org/getProxyConfig

I also updated outdated default address of server in the master branch: https://github.com/alexbers/mtprotoproxy/commit/c89479000fb662c48f0ccc514fdd8112d58d87c7 may be it will help

**Comment by aepot** [comment]
@alexbers looks like the IPs and ports were changed again.

**Comment by alexbers** [comment]
I've rechecked, they are still working. The main IP's also the same: https://github.com/telegramdesktop/tdesktop/blob/master/Telegram/SourceFiles/mtproto/mtproto_dc_options.cpp#L32

**Comment by aepot** [comment]
@alexbers you're right. I've checked another source, the pyrogram lib's docs. Btw, there's no middle proxies with 80 port open anymore in the list by the link you've given above. All the proxies are on 8888.

**Comment by alexbers** [comment]
fixed, thanks.

https://github.com/alexbers/mtprotoproxy/commit/bc841cff482a72472f15a36f86ca01aa11cac58b

**Comment by fromstark** [comment]
Got same problem. I've installed two same copies of proxies on different VPS, first one is working well, but second only works when AD_TAG is empty. What 

TG_MIDDLE_PROXIES_V4 = {
    1: [("149.154.175.50", 8888)], -1: [("149.154.175.50", 8888)],
    2: [("149.154.161.144", 80)], -2: [("149.154.161.144", 8888)],
    3: [("149.154.175.100", 8888)], -3: [("149.154.175.100", 8888)],
    4: [("91.108.4.138", 8888)], -4: [("149.154.165.109", 8888)],
    5: [("91.108.56.135", 8888)], -5: [("91.108.56.135", 8888)],
}

**Comment by majal** [comment]
I use domain names (FQDN) to do [round-robin DNS load balancing](https://developers.cloudflare.com/dns/manage-dns-records/how-to/round-robin-dns/) of multiple proxies with a single domain name.

Everything works, until I enable the AD_TAG.

I've read somewhere that the proxy submits the detected machine IP address to Telegram servers with AD_TAG. Would it be possible to submit a custom domain instead of the detected IP (like in config.py)? Will that make AD_TAG work for high-availability round-robin setups like this?

**Comment by uponatime2019** [comment]
6-2025 this still happen to me (try different VPSs)

**Comment by Umarchik** [comment]
When AD_TAG is enabled, the program tries to connect to Telegram servers via IPv6 — and that's the problem.
You need to either disable IPv6 on the host or at the container level.

In the docker-compose.yml file, add:
`services:
  mtprotoproxy:
    ...
    sysctls:
      - net.ipv6.conf.all.disable_ipv6=1
    ports:
      - "443:443"
`
And remove:
`network_mode: "host"
`
Then rebuild the container — and enjoy! 😄

P.S. Don't forget to update TG_MIDDLE_PROXIES_V4 in the mtprotoproxy.py file.

**Comment by alexbers** [comment]
should be fixed in master, please recheck
